pkg/controller/shoot/shoot_control_reconcile.go

```diff
@@ -55,6 +55,7 @@ func (c *defaultControl) reconcileShoot(o *operation.Operation, operationType ga
 		defaultRetry = 30 * time.Second
 		managedDNS   = o.Shoot.Info.Spec.DNS.Provider != gardenv1beta1.DNSUnmanaged
 		isCloud      = o.Shoot.Info.Spec.Cloud.Local == nil
+		isBigCluster = o.Shoot.GetNodeCount() >= common.EnableHPANodeCount
```

Comment:
No extra function call in the flow is needed, please handle that within the `DeployKubeAPIServer` function itself (as discussed above: move the HPA manifest into the kube-apiserver Helm chart and control via values whether it should get deployed or not).